### PR TITLE
Convert all .rst chapters to the .md syntax

### PR DESCRIPTION
**Pull request by bocharsky-bw**
Fix case discussed in https://github.com/knpuniversity/twig/pull/1#issuecomment-280525836

### CODE REVIEW

File: knpu/basics.rst [hunk 1]
```diff
@@ -8,250 +7,237 @@ easy to learn, powerful and a joy to work with.
...
 with some things called :ref:`functions<twig-functions>` and :ref:`filters<twig-filters>`,
 but this is the most fundamental syntax in Twig.
 
-Looping over Variables
-----------------------
+## Looping over Variables
 
-Next, the ``products`` variable is an array that we need to loop through.
+Next, the `products` variable is an array that we need to loop through.
 Twig comes with a `for`_ tag that is able to loop through items just like
```
Comment (bocharsky-bw) [inline knpu/basics.rst]: This is still an .rst link

File: knpu/extra-credit-tricks-escaping.rst [hunk 2]
```diff
@@ -1,239 +1,229 @@
...
+> use them in this way. See the [use tag][use_tag] for more details.
 
-The Short block Syntax
-----------------------
+## The Short block Syntax
 
-And while we're on this topic, we can make the ``title`` block even shorter
-in ``homepage.twig``:
+And while we're on this topic, we can make the `title` block even shorter
+in `homepage.twig]:
```
Comment (bocharsky-bw) [inline knpu/extra-credit-tricks-escaping.rst]: Missing close quote - the invalid `]` instead

File: knpu/extra-credit-tricks-escaping.rst [hunk 3]
```diff
@@ -1,239 +1,229 @@
...
+is with the `~` character, which concatenates strings in Twig.
 
-    {% block title 'Start looking fly! '~pageData.title %}
+```jinja
+{% block title 'Start looking fly! '~pageData.title %}
+```
 
 You won't see this too often, but it'll come in handy when you need it.
 
 .. _twig-control-whitespace:
```
Comment (bocharsky-bw) [inline knpu/extra-credit-tricks-escaping.rst]: Should be removed

File: knpu/basics.rst [hunk 4]
```diff
@@ -8,250 +7,236 @@ easy to learn, powerful and a joy to work with.
...
+```
+
+The `homepage.php` file is the actual template. It has all the HTML and
+we use `foreach` to loop through them and then `echo` to print out some
 variables:
 
-.. code-block:: html+php
-
-    <!-- templates/homepage.php -->
+```php
```
Comment (MolloKhan) [inline knpu/basics.rst]: I think this should be html+php

File: knpu/basics.rst [hunk 5]
```diff
@@ -8,250 +7,236 @@ easy to learn, powerful and a joy to work with.
...
-    <h1>{{ pageTitle }}</h1>
+```html+jinja
+<!-- templates/homepage.twig -->
+<h1>{{ pageTitle }}</h1>
+```
 
 When we refresh the page, it works! We've just written our first line of Twig!
 Whenever you want to print something, just open Twig with two curly braces,
 write the variable name, then close Twig. We'll get fancier in a little while
 with some things called :ref:`functions<twig-functions>` and :ref:`filters<twig-filters>`,
```
Comment (MolloKhan) [inline knpu/basics.rst]: should we convert those `:ref:'` or they will work just fine ?

File: knpu/basics.rst [hunk 6]
```diff
@@ -8,250 +7,236 @@ easy to learn, powerful and a joy to work with.
...
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+### The "Say Something" Syntax: `{{ ... }}`
 
-The double-curly-brace (``{{``) is always used to print something. If whatever you
+The double-curly-brace (`{{`) is always used to print something. If whatever you
 need to do will result in something being printed to the screen, then you'll
 use this syntax. I call this the "say something" tag, ya know, because it's
 how you "speak" in Twig.
 
-.. _`twig-do-something-syntax`:
```
Comment (MolloKhan) [inline knpu/basics.rst]: Do we just have to remove this references for headers ? `.. _`twig-do-something-syntax`:`

File: knpu/basics.rst [hunk 7]
```diff
@@ -8,250 +7,236 @@ easy to learn, powerful and a joy to work with.
...
+### The "Do Something" Syntax: `{% ... %}`
 
-The "Do Something" Syntax: {% ... %}
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-The curly-percent (``{%``) is the other syntax, which I call the "do something"
+The curly-percent (`{%`) is the other syntax, which I call the "do something"
 syntax. It's used for things like `if`_ and `for`_ tags as well as other things
```
Comment (MolloKhan) [inline knpu/basics.rst]: Those (if, for) still are rst refs :)

File: knpu/basics.rst [hunk 8]
```diff
@@ -8,250 +7,236 @@ easy to learn, powerful and a joy to work with.
...
-.. _`Twig Installation`: http://twig.sensiolabs.org/doc/intro.html#installation
-.. _`for`: http://twig.sensiolabs.org/doc/tags/for.html
-.. _`if`: http://twig.sensiolabs.org/doc/tags/if.html
-.. _`Documentation`: http://twig.sensiolabs.org/documentation
+[installation]: http://twig.sensiolabs.org/doc/intro.html#installation
+[for]: http://twig.sensiolabs.org/doc/tags/for.html
+[if]: http://twig.sensiolabs.org/doc/tags/if.html
+[documentation]: http://twig.sensiolabs.org/documentation
+[twig_control_whitespace]: https://knpuniversity.com/screencast/twig/extra-credit-tricks-escaping#twig-control-whitespace
+g
```
Comment (MolloKhan) [inline knpu/basics.rst]: There is an extra **g** at the end of the file

**Comment by MolloKhan** [comment]
Those files shouldn't have md file extension ?

**Comment by bocharsky-bw** [comment]
@larzuk91 Yes, they have to be `.md`, otherwise, they won't be parsed by GitHub as MarkDown content ;)

File: knpu/basics.rst (hunk 4 above)
Comment (bocharsky-bw) [reply knpu/basics.rst]: you're right, thanks!

File: knpu/basics.rst (hunk 5 above)
Comment (bocharsky-bw) [reply knpu/basics.rst]: Yes, we *should* convert `:ref:`, it's a specific rst syntax only. And I already fixed it in further commits. I suppose you're reviewing a bit outdated diff.

File: knpu/basics.rst (hunk 6 above)
Comment (bocharsky-bw) [reply knpu/basics.rst]: Yes, they won't work in md

File: knpu/basics.rst (hunk 7 above)
Comment (bocharsky-bw) [reply knpu/basics.rst]: The same here, you're reviewing a bit outdated diff ;)

File: knpu/basics.rst (hunk 8 above)
Comment (bocharsky-bw) [reply knpu/basics.rst]: And here too :)

File: knpu/basics.rst (hunk 7 above)
Comment (MolloKhan) [reply knpu/basics.rst]: haha, sorry I didn't notice it was outdated

**Pull request merged**